dask_geopandas/core.py

```diff
@@ -389,6 +390,38 @@ def morton_distance(self, p=15):
 
         return distances
 
+    def geohash(self, string=True, p=12):
+
+        """
+        Calculate geohash based on the middle points of the geometry bounds
+        for a given precision
+
+        Parameters
+        ----------
+        string : bool (default True)
+            to return string or int Geohash
+        p : int (default 12)
+            precision of the string Geohash
+        Returns
+        ----------
+        type : pandas.Series
+            Series containing Geohash
+        """
+
+        if p not in range(1, 13):
+                raise ValueError(
+                    "The Geohash precision only accepts an integer value between 1 and 12"
+                )
+
+        geohashes = self.map_partitions(
+            _geohash,
+            string=string,
+            p=p,
+            meta=pd.Series([], name="geohash", dtype=object),
```

Comment:
Is the `dtype` here dependent on whether or not string or integer values are being returned?

Reply:
good spot! updated